README.md

```diff
@@ -1,2 +1,51 @@
-# spark
-Spark is an implement of Axon kicker.
+# Spark
+Spark is responsible for the following tasks:
+- Constructing CKB transactions to implement Axon’s staking mechanism. The related code can be found in the `tx-builder` directory.  
+- Providing query APIs for Axon’s frontend to display information to end users. For example, details such as stake, delegate, and reward history of users. The related code can be found in the `rpc-client` directory.  
+- Recovering Sparse Merkle Trees (SMTs).  
+
+The corresponding architecture is depicted below.
+![spark_arch](./spark_arch.png)
+
+To learn more about Axon’s staking and kicker, please visit [staking protocol](https://docs.axonweb3.io/staking/staking-protocol).
+We will describe the three tasks mentioned above in detail.
+
+# CKB Transaction Construction
+Transaction construction can be divided into three categories:
+
+## Staker
+These transactions are initiated by stakers who are responsible for paying the gas fees. Examples include increasing or decreasing a staker’s stake.  
+
+## Delegator
+These transactions are initiated by delegators who are responsible for paying the gas fees. Examples include increasing or decreasing a delegator’s delegation to a staker.  
+## Kicker
+Within the Axon ecosystem, a kicker plays the permissionless role to address various critical tasks, including:
```

Comment:
Kicker plays a versatile, permissionless role, taking charge of the following critical tasks: